src/datatypes/clinical.coffee

```diff
@@ -30,7 +30,22 @@ module.exports.ValueSet = class ValueSet
       get: -> true
 
   hasMatch: (code) ->
-    codesInList(toCodeList(code), @codes)
+    codesList = toCodeList(code)
+    # InValueSet String Overload
+    if codesList.length == 1 and typeof codesList[0] is 'string'
+      codes_codeSystem = {}
+      for c2 in @codes
+        if codesList[0] == c2.code
+          # Check if code exists but with different codesystem,
+          if codes_codeSystem[c2.code]? and codes_codeSystem[c2.code] != c2.system
+            throw new Error('Duplicate codes in different code systems in same valueset')
```

Comment:
I suggest changing the wording of this error to use the wording from the CQL spec.  Something like:
`In (valueset) is ambiguous because resolved value set contains codes from multiple code systems.`

We could also put the code in the error message like this, which might be helpful:
`In (valueset) is ambiguous--multiple matches for #{code} found in value set with different code systems.`